### ISSUE TITLE
alpha3 not on npm

### ISSUE THREAD

**Issue by pkra** [mathjax/mathjax-v3#83]
Cf. https://travis-ci.org/mathjax/mathjax-v3/jobs/334943476#L718

**Comment by dpvc** [comment]
I have been looking at this for the last hour and a half, and it looks like I have to change the token (I have that done, but haven't pushed it yet).  Since you set this up, I wanted to check if you thought that would do it? Will I have to make another tag to get it to go again, or should I just `npm publish` by hand?  I know you have done this sort of thing in the past, and thought you might give me a lead on the right thing to do.

**Comment by pkra** [comment]
> Since you set this up, I wanted to check if you thought that would do it? 

I wasn't planning on doing it.

> Will I have to make another tag to get it to go again, or should I just npm publish by hand? 

I would probably just publish it manually. If it goes awry, you can still bump the version and have travis do it to fix.

**Comment by dpvc** [comment]
> I wasn't planning on doing it.

I wasn't asking you to do it, I was asking your advice on whether changing the api key (the token) was what needed to be done ("I wanted to check if you thought **that** would do it", not if "**you** would do it").  I'm sorry I wasn't clearer about it.

> If it goes awry, you can still bump the version and have travis do it to fix.

I didn't want to rely on travis because I wasn't sure that changing the key in the `travis.yml` file was what was needed.  That's why I was asking your advice in the first part of the question.  I didn't want to have to keep trying things that would require bumping versions until I was able to get travis working.  I thought you might be able to tell me if that change would be sufficient.

**Comment by pkra** [comment]
> I wasn't asking you to do it, I was asking your advice on whether changing the api key (the token) was what needed to be done ("I wanted to check if you thought that would do it", not if "you would do it"). I'm sorry I wasn't clearer about it.

Ah, sorry, I misunderstood. The error didn't look like it was about the token (something about not findind the package in the registry). I didn't investigate it so I can't give any advice.

> I didn't want to rely on travis because I wasn't sure that changing the key in the travis.yml file was what was needed. That's why I was asking your advice in the first part of the question. I didn't want to have to keep trying things that would require bumping versions until I was able to get travis working. I thought you might be able to tell me if that change would be sufficient.

Copy that. Sorry for the misunderstanding.

**Comment by dpvc** [comment]
The error message listed:

> NPM API key format changed recently. If your deployment fails, check your API key in ~/.npmrc.
> http://docs.travis-ci.com/user/deployment/npm/

which seemed to indicate that they API key had to be updated.  The link indicates a procedure to get the key in place, which I followed.  But the format seems the same as the original (though the key itself is different), and the documentation seems to indicate that the keys are machine dependent, so I wasn't sure if you had to get a key for the machine where travis is running, or not.  I thought you might be able to shed light on that, since you set up the originals.  

The documentation wasn't clear on what the changes were, and didn't give a migration path, just how to set it up initially, so I did that.  I will push the commit and see.  I suspect this means ALL the repositories will need to have their keys changed.

> The error didn't look like it was about the token (something about not findind the package in the registry).

I think that was because they key wasn't accepted, so we didn't get logged in, and so couldn't access the package.  That is how I read the message

> You must be logged in to publish packages. : mathjax3

in connection with the API key change message.

**Comment by pkra** [comment]
> which seemed to indicate that they API key had to be updated. 

This warning has been there for quite a while (long before v3).

Did you change any permissions on the npm account by any chance? NPM added more security measures recently (as opt-ins).

**Comment by dpvc** [comment]
> Did you change any permissions on the npm account by any chance? 

Not that I'm aware of.  I hadn't even logged into it prior to getting your original message in this issue.  But when I did log in and check the tokens page, there were no active tokens, which seemed to confirm that there was a token issue.  After following the steps in the link about the API change, I now have a token listed on npmjs, so that seems to coincide with the idea that there was a token problem.  But again, these are supposed to be machine-dependent, which makes me concerned that using the token I generated from my machine won't work.  I'm not sure how I would make one for the machine where travis is running, however.
